### PR TITLE
Make sure flipping work, debug and fix CX and CPLY loaders

### PR DESCRIPTION
**Pull request by AnastaZIuk**
`COBJ` and `CSTL` loaders have worked good since now with new flipping dependency, but

`CPLY` crashes, but I have made it doesn't crash and now renders primitives
Even though Chair.ply I have downloaded doesn't appear on screen, it's weird because it seems to me everything needed is already set
`CX` crashes because of dereferencing desc member that is nullptr, it will be fixed soon

I have taken last working commits from your master and added my flipiping one more time

### CODE REVIEW

File: src/irr/asset/CXMeshFileLoader.h [hunk 1]
```diff
@@ -193,11 +195,11 @@ class CXMeshFileLoader : public asset::IAssetLoader
 
 	bool parseDataObjectMaterial(SContext& _ctx, video::SCPUMaterial& material);
 
-	bool parseDataObjectAnimationSet(SContext& _ctx);
+	bool parseDataObjectAnimationSet(SContext& _ctx, orientationFunction performActionBasedOnOrientationSystem);
```
Comment (devshgraphicsprogramming) [inline src/irr/asset/CXMeshFileLoader.h]: use a reference, the functional object is very expensive to pass around by copy

In all honesty I'd prefer a bunch of nice staateless static methods to a lambda

Comment (AnastaZIuk) [reply src/irr/asset/CXMeshFileLoader.h]: yup, it will be fixed

**Comment by devshgraphicsprogramming** [comment]
Status according to conversation with @AnastaZIuk :
- Memory Leak in PLY or some sort of corruption leading to crash
- BAW loader does not respect the flag yet but works.

File: src/irr/asset/CXMeshFileLoader.h (hunk 1 above)
Comment (AnastaZIuk) [reply src/irr/asset/CXMeshFileLoader.h]: @devshgraphicsprogramming  well, how should it look like?
do you mean a static method expecting two lambdas for handling orientation?

**Comment by AnastaZIuk** [comment]
> Status according to conversation with @AnastaZIuk :
> 
> * Memory Leak in PLY or some sort of corruption leading to crash
> * BAW loader does not respect the flag yet but works.

There is BAW Loader left

File: src/irr/asset/CXMeshFileLoader.h (hunk 1 above)
Comment (AnastaZIuk) [reply src/irr/asset/CXMeshFileLoader.h]: @devshgraphicsprogramming  I mean, if there should be such a method, it should have access to `_params`. Should I create a pointer to it as a member? How would you like it to look like?

Comment (AnastaZIuk) [reply src/irr/asset/CXMeshFileLoader.h]: @devshgraphicsprogramming how should I apply flipping on BAW Loader exactly?

Comment (devshgraphicsprogramming) [reply src/irr/asset/CXMeshFileLoader.h]: are there still questions?

**Comment by AnastaZIuk** [comment]
@devshgraphicsprogramming  I think it can be **merged**
everything has been tested in significant detail

the plan is to do BAW later, after writer flags
here are loaders that work great with flipping dependecy

File: src/irr/asset/CXMeshFileLoader.h (hunk 1 above)
Comment (AnastaZIuk) [reply src/irr/asset/CXMeshFileLoader.h]: it's been valid still
> how should I apply flipping on BAW Loader exactly?

**Pull request merged**